protocol/x/clob/keeper/process_operations_stateful_validation.go

```diff
@@ -80,6 +80,25 @@ func (k Keeper) MustFetchOrderFromOrderId(
 	return order
 }
 
+// statefulValidateMatchTaker performs stateful validation on the taker order of a specified match.
+func (k Keeper) statefulValidateMatchTaker(
+	ctx sdk.Context,
+	takerOrder types.Order,
+) error {
+	if takerOrder.GetTimeInForce() == types.Order_TIME_IN_FORCE_IOC {
```

Comment:
Nit: for completeness also block FOK orders as well (use `RequiresImmediateExecution`)